### ISSUE TITLE
shared_vpc example broken for 1.0

### ISSUE THREAD

**Issue by bharathkkb** [terraform-google-modules/terraform-google-project-factory#635]
### TL;DR

[shared_vpc](https://github.com/terraform-google-modules/terraform-google-project-factory/blob/master/examples/shared_vpc) example broken for 1.0. It fails with
```
       Error: Invalid count argument
       
         on ../../../modules/shared_vpc_access/main.tf line 45, in resource "google_compute_subnetwork_iam_member" "service_shared_vpc_subnet_users":
         45:   count    = length(local.subnetwork_api)
       
       The "count" value depends on resource attributes that cannot be determined
       until apply, so Terraform cannot predict how many instances will be created.
       To work around this, use the -target argument to first apply only the
       resources that the count depends on.
```

### Observed behavior

The error occurs due to [this configuration svpc_access submodule](https://github.com/terraform-google-modules/terraform-google-project-factory/blob/master/modules/shared_vpc_access/main.tf#L33-L34) where shared_vpc_subnets is a dynamic value and setproduct need to know the value to compute set and not just the count.

Per https://github.com/hashicorp/terraform/issues/29884 looks like this is expected behavior. 

General pattern is for the host project and network to be created in a separate config than the service project creation, so I am leaning towards moving this into setup for our tests. If there is demand we can evaluate an alternate logic using non computed values.

### Terraform Configuration

```hcl
https://github.com/terraform-google-modules/terraform-google-project-factory/blob/master/examples/shared_vpc/main.tf
```

### Terraform Version

```sh
1.0
```

**Comment by bharathkkb** [comment]
blocking https://github.com/terraform-google-modules/terraform-google-project-factory/pull/614 https://github.com/terraform-google-modules/terraform-google-project-factory/pull/617
